milli/src/search/criteria/asc_desc.rs

```diff
@@ -196,6 +196,44 @@ fn facet_ordered_iterative<'t>(
     Ok(Box::new(number_iter.chain(string_iter).map(Ok)) as Box<dyn Iterator<Item = _>>)
 }
 
+fn facet_extreme_value<'t>(
+    index: &'t Index,
+    rtxn: &'t heed::RoTxn,
+    mut extreme_it: Box<dyn Iterator<Item = heed::Result<RoaringBitmap>> + 't>,
+    field_id: FieldId,
+) -> Result<Option<f64>> {
+    let Some(extreme_value) = extreme_it.next() else { return Ok(None) };
+    let documents = index.documents(rtxn, extreme_value?.into_iter())?;
+    let Some((_, document)) = documents.first() else { return Ok(None) };
+
+    let Some(value) = document.get(field_id) else { return Ok(None) };
+    let Ok(value) = std::str::from_utf8(value) else { return Ok(None) };
```

Comment:
Not sure we want to use this syntax right now. But don't worry we will one day :)

Reply:
Awwwww, that's too bad, I like it 💔 

Any ETA as to when we want to use this? After rustfmt support lands https://github.com/rust-lang/rustfmt/issues/4914 ? Some other thing?

I guess the alternative is the good old:

```rust
let value = if let Ok(value) = std::str::from_utf8(value) { value } else { return Ok(None) };
```

Reply:
(also, I may or may not have smuggled a few `let-else`s in other PRs already 🙊)

Reply:
I would say waiting for the rustfmt issue to be closed and most of the common text editor syntax highlighters to be ready, so something like 6 months or so? What do you think? Is it too much?

Maybe we can plan to do a big pass one day to change all of the `if let else` in to `let else` statements?

Reply:
> I would say waiting for the rustfmt issue to be closed and most of the common text editor syntax highlighters to be ready, so something like 6 months or so? What do you think? Is it too much?

Ah, I was not aware that the syntax highlighters weren't ready (it works for me with vscode and r-a).

OK to wait for rusfmt support for now (although I'm a bit unsure if it will take more or less than six months)..., as we have an "automatic" way to desugar the `let else` to `if let else` at the moment (which is the way we were using up to now).

> Maybe we can plan to do a big pass one day to change all of the if let else in to let else statements?

I'd love for a `cargo clippy --fix` for this one. Maybe it's time I take a look at clippy 🤓

Reply:
removed 🙂